pyproject.toml

```diff
@@ -58,18 +58,32 @@ numpy = "1.23.5"
 protobuf = "3.20.3"
 pandas = "2.0.3"
 
-# Needed to have the latest Concrete Python versions
+# Zama sources
 [[tool.poetry.source]]
 name = "zama-pypi-cpu"
 url = "https://pypi.zama.ai/cpu"
 priority = "explicit"
 
-[tool.poetry.dev-dependencies]
+[[tool.poetry.source]]
+name = "zama-pypi-gpu"
+url = "https://pypi.zama.ai/gpu"
+priority = "explicit"
+
+# GPU
+[tool.poetry.group.gpu]
+optional = true
+
+[tool.poetry.group.gpu.dependencies]
+concrete-python = {version="==2.7.0", source = "zama-pypi-gpu", markers = "sys_platform != 'darwin' and platform_machine != 'arm'"}
```

Comment:
do we want to add the gpu dependency in this exact PR ? feels like it should a PR on its own, with some associated tests